### DISCUSSION TITLE
Im not able to replicate getStaticProps flow using next 13

### DISCUSSION THREAD

**Discussion by bairrada97** [vercel/next.js#44756] [Help] [answered]
my app requires to know the timezone and dates (eg. europe-berlin, 2023-01-10).

on next 12 to make sure I was getting faster load times with high dynamic data I created a route based on user timezone, so if I user was from europe-amsterdam I would redirect him to that route including the today's date. The route would be like this
` /europe-amsterdam/2023-01-10.`

The today's date the data needs to update every 15 seconds while other dates are static and it will not change.

in order to achieve this I was using getStaticProps with a On demand ISR, and on client side I was making fetch calls every 15 seconds. so a user that hit that route will have static data and once it calls the endpoint it revalidates, making every user getting updated data and at same time faster load times because I was not using getServerSideProps.

I'm trying to migrate to next 13 and Im not able to replicate this logic, it states on docs that dynamic routes will always be on request time and not build time making it not possible to replicate what I had on next12. I did not use GetStaticPaths() so im assuming generateStaticParams its not the way to go.

 only thing I did besides getStaticProps that I did on next12 with it was 
 
```
export const getStaticPaths: GetStaticPaths<{ slug: string }> = async () => {
  return {
    paths: [], //indicates that no page needs be created at build time
    fallback: "blocking", //indicates the type of fallback
  };
}
;
```

Im not sure if Im missing something but I would appreciate some guidance on this

**Comment by wwwdepot** [comment, marked as answer]
See https://beta.nextjs.org/docs/upgrade-guide#replacing-fallback

**Reply by bairrada97** [reply]
so by the docs
 "With [dynamicParams](https://beta.nextjs.org/docs/api-reference/segment-config#configdynamicparams) set to true (the default), when a route segment is requested that hasn't been generated, it will be server-rendered and cached as [static data](https://beta.nextjs.org/docs/data-fetching/fundamentals#static-and-dynamic-data-fetches) on success"
 
 I've added `export const dynamicParams = true;`  and I still dont see it cached, I see my data changing every 30 seconds when I refresh the page even tho im saying to revalidate for big amount of time,
 
 Here's my code for [timezone]/[currentDate]/ route

```
export const dynamicParams = true;

const fetchFixtureByDate = async (date: string, timezone: string) => {
  const today = formatDateBasedOnTimezone(new Date(), timezone!);
  const isToday = date === today;
  const res = await fetch(
    `https://outscore.fly.dev/api/v3/fixtures?date=${date}&timezone=${timezone}`,
    { next: { revalidate: 6000 } }
  );

  return await res.json();
};

export default async function CurrentDate({
  params: { timezone, currentDate },
}: {
  params: { timezone: string; currentDate: string };
}) {
  const transformedTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;

  const data = await fetchFixtureByDate(currentDate, transformedTimezone);

  return <Homepage data={data} />;
}

```


Edit: I've fixed it by adding 
```
export async function generateStaticParams() {
  return [
    {
      timezone: "",
      currentDate: "",
    },
  ];
}
```

**Reply by wwwdepot** [reply]
You are missing the empty `generateStaticParams` function
```js
export const dynamicParams = true;
export async function generateStaticParams() {
  return [];
}

... rest of your code
```

**Reply by bairrada97** [reply]
yes exactly!

Thanks for helping
